libs/execution_context/ebpf_core.c

```diff
@@ -2321,6 +2321,11 @@ _ebpf_core_trace_printk(_In_reads_(fmt_size) const char* fmt, size_t fmt_size, i
         return -1;
     }
 
+    // If the provider is not enabled, don't bother with the rest.
+    if (!TraceLoggingProviderEnabled(ebpf_tracelog_provider, EBPF_TRACELOG_LEVEL_INFO, EBPF_TRACELOG_KEYWORD_PRINTK)) {
```

Comment:
There would be a (possibly negligible) perf improvement if you moved this up to 2319 since the check in 2319 is very unlikely whereas this check might be in a relatively hot path.